### PR TITLE
UWP Compat Support

### PR DESCRIPTION
**Pull request by mayankbansal018**
- Moved NetStandard dependency from 1.6.1 to 1.6.0
- Added support for picking up old C++, & C# adapters for UWP
- Added Initialization point for UWP apps

### CODE REVIEW

**Comment by mayankbansal018** [comment]
@dotnet-bot test this please

File: scripts/build/TestPlatform.Settings.targets [hunk 1]
```diff
@@ -25,6 +25,7 @@
     <!-- Disable default inclusion of .resx file. We generate files in the Resources directory only
          if localization is enabled, default inclusion ends up including the generated files by default. -->
     <EnableDefaultEmbeddedResourceItems>false</EnableDefaultEmbeddedResourceItems>
+    <NetStandardImplicitPackageVersion>1.6.0</NetStandardImplicitPackageVersion>
```
[the remark this answers is not in the document]
Comment (mayankbansal018) [reply scripts/build/TestPlatform.Settings.targets]: resolved

File: src/Microsoft.TestPlatform.ObjectModel/Microsoft.TestPlatform.ObjectModel.csproj [hunk 2]
```diff
@@ -30,30 +30,30 @@
...
     <PackageReference Include="System.Runtime.Serialization.Primitives">
-      <Version>4.3.0</Version>
+      <Version>4.1.1</Version>
     </PackageReference>
     <PackageReference Include="System.Xml.XPath.XmlDocument">
       <Version>4.0.1</Version>
     </PackageReference>
-    <PackageReference Include="System.ComponentModel.EventBasedAsync">
-      <Version>4.3.0</Version>
+	<PackageReference Include="System.ComponentModel.EventBasedAsync">
```
[the remark this answers is not in the document]
Comment (mayankbansal018) [reply src/Microsoft.TestPlatform.ObjectModel/Microsoft.TestPlatform.ObjectModel.csproj]: resolved

File: src/testhost.x86/UnitTestClient.cs [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    /// </summary>
+    public static class UnitTestClient
+    {
+        /// <summary>
+        /// Create default UI for UWP app
+        /// </summary>
+        public static void CreateDefaultUI()
+        {
+        }
+
```
[the remark this answers is not in the document]
Comment (mayankbansal018) [reply src/testhost.x86/UnitTestClient.cs]: resolved

File: src/Microsoft.TestPlatform.ObjectModel/Microsoft.TestPlatform.ObjectModel.csproj (hunk 2 above)
[the remark this answers is not in the document]
Comment (mayankbansal018) [reply src/Microsoft.TestPlatform.ObjectModel/Microsoft.TestPlatform.ObjectModel.csproj]: resolved

File: src/Microsoft.TestPlatform.Common/ExtensionFramework/TestPluginDiscoverer.cs [hunk 4]
```diff
@@ -61,46 +61,36 @@ internal class TestPluginDiscoverer
...
-                                Select<IStorageFile, string>(file => file.Name);
+        #region Private Methods
 
-            GetTestExtensionsFromFiles<TPluginInfo, TExtension>(binaries.ToArray(), loadOnlyWellKnownExtensions, pluginInfos);
+        private void AddKnownExtensions(ref IEnumerable<string> extensionPaths)
+        {
+            // For C++ UWP adatper
+            if(File.Exists("Microsoft.VisualStudio.TestTools.CppUnitTestFramework.CppUnitTestExtension.dll"))
+            {
+                extensionPaths = extensionPaths.Concat(new List<string> { "Microsoft.VisualStudio.TestTools.CppUnitTestFramework.CppUnitTestExtension.dll" });
```
[the remark this answers is not in the document]
Comment (mayankbansal018) [reply src/Microsoft.TestPlatform.Common/ExtensionFramework/TestPluginDiscoverer.cs]: resolved

File: src/Microsoft.TestPlatform.Common/ExtensionFramework/TestPluginDiscoverer.cs [hunk 5]
```diff
@@ -61,46 +61,36 @@ internal class TestPluginDiscoverer
...
 
-                                ).
-                                Select<IStorageFile, string>(file => file.Name);
+        #region Private Methods
 
-            GetTestExtensionsFromFiles<TPluginInfo, TExtension>(binaries.ToArray(), loadOnlyWellKnownExtensions, pluginInfos);
+        private void AddKnownExtensions(ref IEnumerable<string> extensionPaths)
+        {
+            // For C++ UWP adatper
+            if(File.Exists("Microsoft.VisualStudio.TestTools.CppUnitTestFramework.CppUnitTestExtension.dll"))
```
[the remark this answers is not in the document]
Comment (mayankbansal018) [reply src/Microsoft.TestPlatform.Common/ExtensionFramework/TestPluginDiscoverer.cs]: resolved

File: src/Microsoft.TestPlatform.Common/ExtensionFramework/TestPluginDiscoverer.cs [hunk 6]
```diff
@@ -13,12 +13,33 @@ namespace Microsoft.VisualStudio.TestPlatform.Common.ExtensionFramework
...
     /// Discovers test extensions in a directory.
     /// </summary>
     internal class TestPluginDiscoverer
     {
+        private IFileHelper fileHelper;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="TestPluginDiscoverer"/> class. 
+        /// </summary>
+        public TestPluginDiscoverer() : this(new Microsoft.VisualStudio.TestPlatform.Utilities.Helpers.FileHelper())
```
Comment (singhsarab) [inline src/Microsoft.TestPlatform.Common/ExtensionFramework/TestPluginDiscoverer.cs]: >Microsoft.VisualStudio.TestPlatform.Utilities.Helpers [](start = 49, length = 53)

Why not using?

File: src/Microsoft.TestPlatform.Common/ExtensionFramework/TestPluginDiscoverer.cs [hunk 7]
```diff
@@ -61,46 +82,36 @@ internal class TestPluginDiscoverer
 
             var pluginInfos = new Dictionary<string, TPluginInfo>();
 
-#if !WINDOWS_UAP
+            // C++ UWP adapters do not follow TestAdapater naming convention, so making this exception
+            if (!extensionPaths.Any())
+            {
+                AddKnownExtensions(ref extensionPaths);
```
Comment (singhsarab) [inline src/Microsoft.TestPlatform.Common/ExtensionFramework/TestPluginDiscoverer.cs]: Nit: this.

File: test/testhost.UnitTests/UnitTestClientTests.cs [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+    [TestClass]
+    public class UnitTestClientTests
+    {
+        [TestMethod]
+        public void SplitArgumentsShouldHonorDoubleQuotes()
+        {
+            var argument = "--port 8080 --endpoint 127.0.0.1:8020 --diag \"abc txt\"";
+            string[] argsArr = UnitTestClient.SplitArguments(argument);
+
+            Assert.AreEqual(argsArr.Length, 6);
```
Comment (singhsarab) [inline test/testhost.UnitTests/UnitTestClientTests.cs]: Nit: CollectionAssert, maybe?

File: test/TestAssets/SimpleTestProject/SimpleTestProject.csproj [hunk 9]
```diff
@@ -8,6 +8,7 @@
   <PropertyGroup>
     <AssemblyName>SimpleTestProject</AssemblyName>
     <TargetFrameworks>netcoreapp1.0;netcoreapp1.1;netcoreapp2.0;net451</TargetFrameworks>
+    <NetStandardImplicitPackageVersion>1.6.0</NetStandardImplicitPackageVersion>
```
Comment (singhsarab) [inline test/TestAssets/SimpleTestProject/SimpleTestProject.csproj]: >NetStandardImplicitPackageVersion [](start = 5, length = 33)

We should not need to change the TestAssets.

File: test/Microsoft.TestPlatform.Common.UnitTests/ExtensionFramework/TestPluginDiscovererTests.cs [hunk 10]
```diff
@@ -125,6 +127,21 @@ public void GetTestExtensionsInformationShouldNotAbortOnFaultyExtensions()
             Assert.That.DoesNotThrow(() =>this.testPluginDiscoverer.GetTestExtensionsInformation<FaultyTestExecutorPluginInformation, ITestExecutor>(pathToExtensions, loadOnlyWellKnownExtensions: true));
         }
 
+        [TestMethod]
+        public void GetTestExtensionsInformationShouldAddUWPCppAdatersIfTheyExists()
```
Comment (singhsarab) [inline test/Microsoft.TestPlatform.Common.UnitTests/ExtensionFramework/TestPluginDiscovererTests.cs]: Add the test for the other dll or include it in this one itself.

**Review by singhsarab — APPROVED** [review]
Looks good with suggestions.

File: scripts/build/TestPlatform.Settings.targets [hunk 11]
```diff
@@ -25,6 +25,9 @@
     <!-- Disable default inclusion of .resx file. We generate files in the Resources directory only
          if localization is enabled, default inclusion ends up including the generated files by default. -->
     <EnableDefaultEmbeddedResourceItems>false</EnableDefaultEmbeddedResourceItems>
+    <!-- .NetStandardLibrary1.6.0 brings in dependencies which are supported in UWP,
+         any higher version will bring in dependencies which are not supported by current UWP version -->
```
Comment (smadala) [inline scripts/build/TestPlatform.Settings.targets]: What is current UWP version.

File: src/Microsoft.TestPlatform.ObjectModel/Navigation/DiaSession.cs [hunk 12]
```diff
@@ -95,6 +95,10 @@ public INavigationData GetNavigationDataForMethod(string declaringTypeName, stri
         private static ISymbolReader GetSymbolReader(string binaryPath)
         {
             var pdbFilePath = Path.ChangeExtension(binaryPath, ".pdb");
+
+            // For remote scenario, also look for pdb in current directory, (esp for UWP)
+            // The alternate search path should be an input from Adapters, but since it is not so currently adding a HACK
```
Comment (smadala) [inline src/Microsoft.TestPlatform.ObjectModel/Navigation/DiaSession.cs]: Look in `searchPath` too if not found in current directory?

File: test/Microsoft.TestPlatform.Common.UnitTests/ExtensionFramework/TestPluginDiscovererTests.cs (hunk 10 above)
Comment (mayankbansal018) [reply test/Microsoft.TestPlatform.Common.UnitTests/ExtensionFramework/TestPluginDiscovererTests.cs]: resolved

File: scripts/build/TestPlatform.Settings.targets (hunk 11 above)
Comment (mayankbansal018) [reply scripts/build/TestPlatform.Settings.targets]: UWP is on NetStandard 1.4, and on NetStandardLibrary1.6.0

File: src/Microsoft.TestPlatform.ObjectModel/Navigation/DiaSession.cs (hunk 12 above)
Comment (mayankbansal018) [reply src/Microsoft.TestPlatform.ObjectModel/Navigation/DiaSession.cs]: I agree, we will bring this in later

**Review by smadala — APPROVED** [review]

**Pull request merged**